create_language_model.py

```diff
@@ -0,0 +1,241 @@
+
+english_letters = [
+  'A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z',
+  'a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
+
+def first_letter_to_uppercase(string):
+    return string[0].upper() + string[1:]
+  
+def normalize_charset_name(string):
+    return first_letter_to_uppercase(string).replace('-', '_');
+
+try:
+    file_name = sys.argv[1]
+    charset_name = sys.argv[2]
+    keep_english_letter = sys.argv[3]
+except:
+    print "Usage: %s input_file_in_utf_8 charset_name keep_english_letter" % sys.argv[0]
+    sys.exit(1)
+ 
+keep_english_letter = True if keep_english_letter.lower() in ['1', 'true', 'yes', 'y'] else False
+
+# ***************************************************** CREATING CHARSMAP TABLE FOR GIVEN CHARSET ***************************************************** #
+charsmap_table = OrderedDict([])
+counter = NULL
+xx_counter = 0
+with codecs.open('CharsetsTabs.txt', 'r', encoding='utf-8') as f: # File 'CharsetsTabs.txt' must be included with this script!
+    for line in f:						  # It contains tables of common character sets like the Latin-2, Windows-1250, ...)
+	if charset_name.lower() in line.lower():  
+	    chars = []
+	    counter = 0
+	    continue
+	if counter > 16:
+	    break
+	if counter != NULL:
+	    counter += 1
+	if counter > 1:
+	    chars = line.split()[1:]
+	    for char in chars:
+		if char == 'XX':
+		    charsmap_table['XX' + str(xx_counter)] = CTRL # The 'XX' in the charset table indicates a nonexisting character
+		    xx_counter += 1
+		elif char == 'lf':
+		    charsmap_table['\r'] = CRLF
+		elif char == 'cr':
+		    charsmap_table['\n'] = CRLF
+		elif char == 'sp':
+		    charsmap_table[' '] = SYMBOL
+		elif len(char) == 1:
+		    if char.isdigit():
+			charsmap_table[char] = DIGIT
+		    elif char.isalpha():
+			if not keep_english_letter and char in english_letters:
+			    charsmap_table[char] = CTRL
+			else:
+			    charsmap_table[char] = ALPHA
+		    else:
+			charsmap_table[char] = SYMBOL
+		else:
+		    charsmap_table[char] = CTRL
+    f.close()
+    
+if counter == NULL:
+    print "Wrong/unknown charset name! Please check the 'CharsetsTabs.txt' file."
+    sys.exit(1)
+
+# *************************************************************** CHARS FREQUENCY TABLE *************************************************************** #
+letters_table = []
+for char in charsmap_table:
+    if charsmap_table[char] == ALPHA:
+	letters_table.append(char)
+
+chars_frequency_table = OrderedDict([])
+for char in charsmap_table: 
+    chars_frequency_table[char] = 0
+
+lang_name = os.path.splitext(os.path.basename(file_name))[0] 
+print "Creating chars frequency table for %s language" % lang_name
+with codecs.open(file_name, 'r', encoding='utf-8') as f:
+    for line in f:
+	for char in line:
+	    if char in chars_frequency_table:
+		chars_frequency_table[char] += 1
+    f.close()
```

Comment:
Thank you for info, I'm really new in python. My "native" languages are C++, C and assembler.